### PR TITLE
Improve application process

### PR DESCRIPTION
**Pull request by lisbethmarianne**
add questions to application form
add terms and conditions to application form
add validations for new fields in the application form
add option for admin to view and delete individual applications
add new fields to the CSV download

Issues #83 and #111 

Please feel free to comment!

### CODE REVIEW

**Comment by lisbethmarianne** [comment]
Any suggestions for this PR?

File: app/views/applications/new.html.erb [hunk 1]
```diff
@@ -40,6 +55,30 @@
 
   <div class="form_field">
     <p>
+      <%= f.check_box :terms_and_conditions %>
+      <%= f.label :terms_and_conditions, "Terms and Conditions" %>
```
Comment (alicetragedy) [inline app/views/applications/new.html.erb]: just a minor improvement: could you change this to either
"Terms and Conditions:"
or
"I agree with the following Terms and Conditions:"

basically to make the checkbox more understandable :)

File: app/views/applications/new.html.erb [hunk 2]
```diff
@@ -40,6 +55,30 @@
...
+    </p>
+    <p>
+      This diversity program is aimed to help underrepresented groups in tech to attend and lowering the barriers for them. This includes but isn’t limited to: women, people of color, LGBTQIA+ people, and disabled people. Your privacy, details, answers remain protected, we will raffle out the tickets. We will not share any data with those, who sponsor these tickets. This application is for one conference ticket only. The ticket includes
```
Comment (alicetragedy) [inline app/views/applications/new.html.erb]: Could you change to: 
Your privacy, details and answers remain protected; we will raffle out the tickets. We will not share any data with those who sponsor these tickets.

File: app/views/applications/show.html.erb [hunk 3]
```diff
@@ -10,5 +22,8 @@
   <%= @application.email %>
 </p>
 
+<p>
+<%= link_to 'Delete this application', event_application_path(@event.id, @application.id), method: :delete, data: { confirm: "You sure?" } %>
```
Comment (alicetragedy) [inline app/views/applications/show.html.erb]: Maybe a more thorough confirmation message, like "Are you sure you want to delete this application?"

File: app/views/events/admin_show.html.erb [hunk 4]
```diff
@@ -15,8 +15,9 @@
 
 <% @event.applications.each do |application| %>
   <ul class="row">
-    <li><%= application.name %></li>
+    <li><%= link_to application.name, event_application_path(@event, application) %></li>
     <li><%= application.email %></li>
+    <li><%= link_to 'Delete', event_application_path(@event.id, application.id), method: :delete, data: { confirm: "You sure?" } %></li>
```
Comment (alicetragedy) [inline app/views/events/admin_show.html.erb]: improve confirmation message (see comment for `app/views/applications/show.html.erb`)

**Comment by alicetragedy** [comment]
This looks good to me @lisbethmarianne. I have added some comments regarding wording, nothing major. 
I've also noticed that the styling in the event view (where all applications are listed) looks a bit wonky, but I assume we can put that in a separate issue.

**Comment by lisbethmarianne** [comment]
Thanks for the review @alicetragedy

**Pull request merged**